File: easy/53-Maximum_Subarray.py
def max_sub_array(self, nums: List[int]) -> int:
    ans = nums[0]
    cur_sum = 0
    
    for val in nums:
        if cur_sum < 0:
            cur_sum = 0
        cur_sum += val
        ans = max(ans, cur_sum)
    return ans

File: easy/53-Maximum_Subarray.cpp
int maxSubArray(vector<int>& nums) {
	int ans = nums[0];
	int curSum = 0;
	
	for(int a = 0; a < nums.size(); ++a) {
		if(curSum < 0)
			curSum = 0;
		curSum += nums[a];
		ans = max(ans, curSum);
	}
	return ans;
}